presto-docs/src/main/sphinx/security/salesforce.rst

```diff
@@ -0,0 +1,78 @@
+Salesforce data. It's simply a means by which users can authenticate to Presto, similar
+to :doc:`ldap` or :doc:`password-file`.
+
+Salesforce Authenticator Configuration
+--------------------------------------
+
+Enable Salesforce authentication by creating an
+``etc/password-authenticator.properties`` file on the coordinator:
+
+.. code-block:: none
+
+    password-authenticator.name=salesforce
+    salesforce.allowed-organizations=allowed salesforce org id's, or the word all
+
+The following configuration properties are available:
+
+====================================   ==============================================
+Property                               Description
+====================================   ==============================================
+``salesforce.allowed-organizations``   Comma separated list of 18 character Salesforce.com
+                                       Organization ID's for a second simple layer of security.
+                                       This option can be explicitly ignored using ``all``
+                                       here, which will bypass any check of the authenticated
+                                       user's Salesforce Organization Id.
+
+``salesforce.cache-size``              Maximum size of the cache that holds authenticated users.
+                                       Defaults to ``4096``.
+
+``salesforce.cache-expire-duration``   Expire duration for an entry in cache since last write.
+                                       Defaults to ``2m``.
+====================================   ==============================================
+
+Salesforce Concepts
+-------------------
+
+.. _security token:
+
+Security Token
+^^^^^^^^^^^^^^
+
+Credentials are a user's Salesforce username and password if Presto is connecting from a whitelisted
+IP, or username and password / security token concatenation otherwise.  For example, if the coordinator
+is not whitelisted, and your password is ``password`` and security token is ``token``, use ``passwordtoken``
+as your Presto password.
+
+To learn more about the Salesforce security token, see the `reset security token documentation
+<https://help.salesforce.com/articleView?id=user_security_token.htm&type=5>`_.
+
+To learn more about whitelisting an IP range, see the `set trusted IP ranges documentation
+<https://help.salesforce.com/articleView?id=security_networkaccess.htm&type=5>`_.  Use this to
+procedure to trust the public IP for the Presto coordinator.
+
+Salesforce Organization Id's
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is a secondary layer of security beyond simply providing credentials, which is that
+the admin can configure one or more 18 character Salesforce.com Organization ID's to which the
+user must belong.  When the user authenticates, the Salesforce API returns the 18 character
+Salesforce.com Organization ID for the user.  Presto's Salesforce authenticator will ensure this
```

Comment:
organization

Reply:
The Presto Salesforce authenticator ensures that the ID matches on of the IDs configured in ..

Reply:
Will update the sentence.  I capitalize Organization within the context of Salesforce.com Organization ID because that's how it appears in the Salesforce UI.  Hopefully this is ok.